### DISCUSSION TITLE
Creating a python plugin `ModuleNotFoundError: No module named 'rzlang'`

### DISCUSSION THREAD

**Discussion by kuterd** [rizinorg/rizin#4585] [Q&A]
Hello everyone, I am trying to create a plugin for a custom architecture.
I am following https://book.rizin.re/src/plugins/python.html 

```
 rizin -I mycpu.py 
Traceback (most recent call last):
  File "/home/normal/projects/rizin_test/mycpu.py", line 1, in <module>
    import rzlang
ModuleNotFoundError: No module named 'rzlang'
```

Thanks

**Comment by konchunas** [comment]
That is exactly my problem. I am on Mac OS with `rizin` installed using `brew` in case that matters.

**Comment by Ganesh-403** [comment]
@kuterd `rzlang` isn't a package that you install separately from PyPI—it's provided by Rizin's **Python language plugin/bindings**. The handbook examples assume those bindings are already available in your Python environment, but they don't explicitly mention that prerequisite. :contentReference[oaicite:0]{index=0}

A couple of things to check:

1. **Verify that the Python language plugin is available**

   Run:

   ```bash
   rizin -L
   ```

   or start Rizin with debug logging:

   ```bash
   RZ_DEBUG=yes rizin
   ```

   and look for something like:

   ```
   lang_python3.so
   ```

   being loaded. If the Python language plugin isn't present, `rzlang` won't be available. :contentReference[oaicite:1]{index=1}

2. **How was Rizin installed?**

   On macOS, if you installed Rizin via Homebrew, could you also share:

   ```bash
   brew info rizin
   python3 --version
   which python3
   ```

   Homebrew installs its own Python and manages `site-packages`, so it's possible that Rizin's Python bindings were built against a different Python installation than the one executing your script. :contentReference[oaicite:2]{index=2}

3. **Can Python import `rzlang` directly?**

   Outside of Rizin, try:

   ```bash
   python3 -c "import rzlang"
   ```

   If that also raises `ModuleNotFoundError`, then the bindings are either not installed or not on your Python search path.

From what I can tell, this is more likely an installation/environment issue than a problem with your plugin itself. The handbook's example is correct, but it assumes the `rzlang` bindings are already available. :contentReference[oaicite:3]{index=3}

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
